src/main/java/frc/robot/RobotContainer.java

```diff
@@ -105,26 +100,32 @@ private void configureBindings() {
         new DefaultDrive(drive,
             () -> getJoystickInput(driverController, Constants.Controller.LEFT_Y_AXIS),
             () -> getJoystickInput(driverController, Constants.Controller.LEFT_X_AXIS),
-            () -> getJoystickInput(driverController, Constants.Controller.RIGHT_X_AXIS)));
+            () -> getJoystickInput(driverController, Constants.Controller.RIGHT_X_AXIS),
+            () -> getDriveSpeedScale()
 
-    driverController.start().onTrue(new InstantCommand(() -> drive.zeroGyroscope(), drive));
+            )
+    );
+
+    shooter.setDefaultCommand(new IntakeNote(shooter, undertakerSubsystem, () -> arm.isAtHomePostion() && intakeEnabled));
 
-    operatorController.b().onTrue(
-        new MoveToPosition(arm, Constants.Arm.AMP_POSITION));
+    driverController.start().onTrue(new InstantCommand(() -> drive.zeroGyroscope(), drive));
+    driverController.rightTrigger().onTrue(new SpinUpAndShoot(shooter, undertakerSubsystem, () -> arm.isAtAmpPosition()));
 
-    operatorController.x().onTrue(new MoveToHome(arm));
+    operatorController.leftTrigger().onTrue(new MoveToPosition(arm, Constants.Arm.AMP_POSITION));
+    operatorController.rightTrigger().onTrue(new MoveToHome(arm));
 
-    // driverController.a().onTrue(new SpeakerScore(drive, autoGenerator));
-    
-    // SHOOTER [leftTrigger -> intakes note; rightTrigger -> shoots]
-    driverController.leftTrigger().whileTrue(new IntakeNote(shooter, undertakerSubsystem));
-    driverController.rightTrigger().onTrue(new SpinUpAndShoot(shooter));
+    operatorController.a().onTrue(new InstantCommand(() -> intakeEnabled = true));
+    operatorController.y().onTrue(new InstantCommand(() -> intakeEnabled = false));
   }
 
   private double getJoystickInput(CommandXboxController stick, int axe) {
     return -MathUtils.getDeadzoneAdjustedInput(stick.getRawAxis(axe));
   }
 
+  private double getDriveSpeedScale() {
```

Comment:
I like how this works but i think we should organize it differently (i dont exactly know how yet). We will do that after driver tryouts